unitTests/magicbox/QueryProcessorTest.ts

```diff
@@ -66,26 +66,39 @@ export function QueryProcessorTest() {
         const stringArraysInResolvedOrder = [['abcd', 'efgh'], ['ijkl'], ['mnop', 'qrst'], ['uvwx']];
         let stringPromisesInResolvedOrder: (string[] | Promise<string[]>)[];
         let stringPromisesInParameterOrder: (string[] | Promise<string[]>)[];
+        let queryProcessor: QueryProcessor<string>;
+        let returnedPromise: Promise<IQueryProcessResult<string>>;
 
         beforeEach(() => {
           stringPromisesInResolvedOrder = [
-            stringArraysInResolvedOrder[0],
+            wait(resolveTimes[3], stringArraysInResolvedOrder[0]),
```

Comment:
By moving this line to the top, does the variable name `stringPromisesInResolvedOrder` still make sense? Seems the promises are no longer in resolved order?

Reply:
You're absolutely right, the resolved order HAS to be the parameter order, so there's no different now